### ISSUE TITLE
[2.8] drop event no longer can get the entity that dropped the item

### ISSUE THREAD

**Issue by Fusezion** [SkriptLang/Skript#6217]
### Skript/Server Version

```markdown
[13:53:35 INFO]: [Skript] Skript's aliases can be found here: https://github.com/SkriptLang/skript-aliases
[13:53:35 INFO]: [Skript] Skript's documentation can be found here: https://docs.skriptlang.org/
[13:53:35 INFO]: [Skript] Skript's tutorials can be found here: https://docs.skriptlang.org/tutorials
[13:53:35 INFO]: [Skript] Server Version: git-Paper-318 (MC: 1.20.2)
[13:53:35 INFO]: [Skript] Skript Version: 2.8.0-dev (selfbuilt-unknown)
[13:53:35 INFO]: [Skript] Installed Skript Addons:
[13:53:35 INFO]: [Skript]  - skript-gui v1.3 (https://github.com/APickledWalrus/skript-gui)
[13:53:35 INFO]: [Skript]  - skript-reflect v2.4-beta1 (https://github.com/SkriptLang/skript-reflect)
[13:53:35 INFO]: [Skript]  - SkBee v3.0.0-beta2 (https://github.com/ShaneBeee/SkBee)
[13:53:35 INFO]: [Skript] Installed dependencies: None
```

### Bug Description

When attempting to get the entity who dropped an item in a drop event or the item dropped in a drop event skript will now cause an error to be sent as there's multiple entities within this event.

### Expected Behavior

Skript to not error and provide me my requested entity. Tho that's not gonna happen the error makes sense but something else needs to be done now.

### Steps to Reproduce

run this code while using a 2.8-dev (or any newer dev/feature) build
```
on drop:
	broadcast event-entity
	broadcast event-dropped item
```

### Errors or Screenshots

![image](https://github.com/SkriptLang/Skript/assets/81952476/c6def08e-9932-4e7d-9482-e4511e57a7ae)

### Other

Honestly there's nothing much that could be done about the errors, the whole point of the error was to make it known, however this is a very large breaking change in the fact it would likely require some other way of getting these values.

This is on a freshly built dev/feature with master branch merged into it.

### Agreement

- [X] I have read the guidelines above and affirm I am following them with this report.

**Comment by GodModed** [comment]
I don't know about `event-entity` but you use `event-item` for getting the dropped item.

**Comment by sovdeeth** [comment]
Have you tried with #6122?

**Comment by Fusezion** [comment]
> Have you tried with #6122?

I have not yet I'll do so after work

**Comment by Fusezion** [comment]
> I don't know about `event-entity` but you use `event-item` for getting the dropped item.

`event-dropped item` refers to the item entity `event-item` is the itemstack

**Comment by Fusezion** [comment]
> Have you tried with #6122?

Alright after trying again with a build using the exact same code it works as the build appears to intend it to now. Since `event-dropped item` was defined we're looking for a type of it there's only 1

However since there's two `event-entity` that line will still error see image for detail. 2.7 and below `event-entity` refers to the entity dropping the item. This isn't solved by `event-player` only as there's also an entity version of this event

![image](https://github.com/SkriptLang/Skript/assets/81952476/328e195b-3183-4887-bed9-529e61cd70aa)

**Comment by sovdeeth** [comment]
Good point
We'll have to figure out how to address that
Thanks for investigating

**Comment by sovdeeth** [comment]
@TheLimeGlass do you have any ideas for fixing this? Currently I think the best option may just be to revert #5769 for now, as I can't think of any good ways to get around this.

**Comment by Fusezion** [comment]
Not lime but I do want to say at least this, I do not believe reverting it is the best call to make.
This feature targets 2.8 and was made to prevent unsafe event-values that wouldn't promise intended design.
This was known to cause issues someway somehow in many other events there's no exact way to test for it without a public test for it.

I believe the best bet is just add some type of new expression or something similar that can be generalized to make sense in other events that show up with this issue.
I haven't tested however `event-living entity` should resolve the error. Since item entity isn't living classified. I believe it's better to update it with a custom error that points to `event-living entity` or some other expression.

I don't love the idea of pointing towards another event-value but it's better to do it than revert a needed change.

**Comment by sovdeeth** [comment]
> Not lime but I do want to say at least this, I do not believe reverting it is the best call to make. This feature targets 2.8 and was made to prevent unsafe event-values that wouldn't promise intended design. This was known to cause issues someway somehow in many other events there's no exact way to test for it without a public test for it.
> 
> I believe the best bet is just add some type of new expression or something similar that can be generalized to make sense in other events that show up with this issue. I haven't tested however `event-living entity` should resolve the error. Since item entity isn't living classified. I believe it's better to update it with a custom error that points to `event-living entity` or some other expression.
> 
> I don't love the idea of pointing towards another event-value but it's better to do it than revert a needed change.

I agree that avoiding the event-value ambiguity is a very needed change, but the current implementation forces weird event-value choices like event-living entity, as you said, or having to make a specific expression for each possible collision.

I suppose making specific expressions for each is the most reliable (and understandable) option for users, but I feel that something like this, where event-entity and event-dropped item are rather clearly different, should still be allowed.

If anyone else has opinions, please chime in!

**Comment by Fusezion** [comment]
Closing this as it seems to no longer be prevalent on newer versions
